### DISCUSSION TITLE
Taiko login information

### DISCUSSION THREAD

**Discussion by Kuryuuki** [getgauge/taiko#2351] [Development]
I noticed that taiko always resets browser information like history, login, etc. Is there a way to not reset this chromium information even if taiko closes and reopens? I always wanted to keep the login on certain sites, without logging in and logging out

**Comment by NivedhaSenthil** [comment]
you could try creating a profile and launch Taiko by passing that as an browser argument for `--user-data-dir`

**Reply by Kuryuuki** [reply]
what do you mean create a profile?

For example: I wanted to log in to a twitter account and always keep logged in to that same account, even if the taiko script is restarted, so that the server doesn't keep identifying multiple logins whenever the program closes and opens

**Reply by sriv** [reply]
@NivedhaSenthil was referring to https://chromium.googlesource.com/chromium/src/+/HEAD/docs/user_data_dir.md

Basically you tell taiko to lauch chromium with a specific user data dir, that is not reset between launches. This way your cookies, preferences etc are saved and the next time you launch the browser with the same profile, the information is restored. In this case, you may login to twitter using the same chromium profile and tell twitter to "remember you" . The next time taiko launches chromium with the same profile, it should have the user logged in twitter. Does this make sense?

**Reply by Kuryuuki** [reply]
@sriv totally sense, but I a newbie! racking my brain to learn the basics. How do I use this "--user-data-dir" command in the taiko code and how do I get this "user"? 
Can I get your Discord or something like that?

**Reply by zabil** [reply]
@Kuryuuki you can refer to this issue https://github.com/getgauge/taiko/issues/983 for code samples
